server/src/main/java/org/candlepin/resource/OwnerProductResource.java

```diff
@@ -221,8 +229,8 @@ public Product createProduct(@PathParam("owner_key") String ownerKey, Product pr
     @Path("/{product_id}")
     @Produces(MediaType.APPLICATION_JSON)
     public Product updateProduct(
-        @PathParam("owner_key") String ownerKey,
-        @PathParam("product_id") String productId,
+        @Verify(Owner.class) @PathParam("owner_key") String ownerKey,
+        @Verify(Product.class) @PathParam("product_id") String productId,
```

Comment:
Similar to create, probably should be super admin only as far as I can think ahead right now.